shacl12-inf-rules/index.html

```diff
@@ -160,444 +223,1890 @@
+              or there is an [=assignment variable=] in an [=assignment=] at position 
+              <em>j</em> where <em>i &gt; j</em>.
+            </li>
+          </ul>
+          <p>
+            A [=rule set=] is "well-formed" if and only if all of the [=rules=] of the
+            rule set are "well-formed".
+          </p>
+        </section>
+    </section>
+
+    <section id="concrete-syntax">
+      <h2>Concrete Syntax forms for Shapes Rules</h2>
+      <ul>
+        <li><a href="rdf-rules-syntax">RDF Rules Syntax</a></li>
+        <li><a href="compact-rules-syntax">Compact Rules Syntax</a></li>
+      </ul>
+
+      <p class="note">
+        The compact syntax has an equivalent RDF syntax form.
+        Well-formed RDF syntax can be translated to the compact syntax.
+      </p>
+
+      <p class="ednote">
+        Change the name away from "compact" and leave that free for SHACL-C.
+      </p>
+
+      <p>Compact:</p>
+<pre class="example-rules">
+PREFIX : &lt;http://example/>
+
+DATA { :x :p 1 ; :q 2 . }
+
+RULE { ?x :bothPositive true . } 
+WHERE { ?x :p ?v1  FILTER ( ?v1 &gt; 0 )  ?x :q ?v2  FILTER ( ?v2 &gt; 0 )  }
+
+RULE { ?x :oneIsZero true . } 
+WHERE { ?x :p ?v1 ;  :q ?v2  FILTER ( ( ?v1 = 0 ) || ( ?v2 = 0 ) )  }
+</pre>
+
+<p>RDF:</p>
+
+<pre class="example-rules">
+PREFIX :       &lt;http://example/&gt;
+PREFIX rdf:    &lt;http://www.w3.org/1999/02/22-rdf-syntax-ns#&gt;
+PREFIX sh:     &lt;http://www.w3.org/ns/shacl#&gt;
+PREFIX sparql: &lt;http://www.w3.org/ns/sparql#&gt;
+
+:ruleSet-1
+  rdf:type sh:RuleSet;
+  sh:data (
+    &lt;&lt;( :x :p 1 )&gt;&gt;
+    &lt;&lt;( :x :q 2 )&gt;&gt;
+  );
+  sh:ruleSet (
+    [
+      rdf:type sh:Rule;
+      sh:head (
+        [ sh:subject [ sh:var "x" ] ; sh:predicate :bothPositive; sh:object true ]
+      )
+      sh:body (
+        [ sh:subject [ sh:var "x" ]; sh:predicate :p; sh:object [ sh:var "v1" ] ]
+        [ sh:expr [ sparql:greaterThan ( [ sh:var "v1" ] 0 ) ] ]
+        [ sh:subject [ sh:var "x" ] ; sh:predicate :q; sh:object [ sh:var "v2" ] ]
+        [ sh:expr [ sparql:greaterThan ( [ sh:var "v2" ] 0 ) ] ]
+      );
+    ]
+    [
+      rdf:type sh:Rule;
+      sh:head (
+        [ sh:subject [ sh:var "x" ] ; sh:predicate :oneIsZero ; sh:object true ]
+      )
+      sh:body (
+        [ sh:subject [ sh:var "x" ] ; sh:predicate :p ; sh:object [ sh:var "v1" ] ]
+        [ sh:subject [ sh:var "x" ] ; sh:predicate :q ; sh:object [ sh:var "v2" ] ]
+        [ sh:expr [ sparql:function-or (
+              [ sparql:equals ( [ sh:var "v1" ] 0 ) ]
+              [ sparql:equals ( [ sh:var "v2" ] 0 ) ]
+            ) ]
+        ]
+      );
+    ]
+  ) .
+</pre>
+
+      <section id="rdf-rules-syntax">
+        <h3>RDF Rules Syntax</h3>
+
+        <div class="ednote">
+          <p>Well-formed ness:</p>
+          <ul>
+            <li>All RDF list are well-formed</li>
+            <li>exactly one of subject - predicate - object per body of head
```

Comment:
To make it less likely to be interpreted as having a period where I'm adding a comma.
```suggestion
            <li>exactly one of subject - predicate - object, per body of head
```

Reply:
Change made.

However, blocking merging for something in an editors note is taking up time.